misc/config_tools/upgrade_scenario_xml

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+function usage() {
+	echo "Usage: $0 <scenario XML> [<version of the given scenario XML>]"
+	echo "Upgrade the given scenario XML to align with the current version of ACRN."
+	echo ""
+	echo "Example: $0 logical_partition.xml v2.3"
+	echo "    The output file is logical_partition-v2.3.xml"
+}
+
+file=$1
+ver=$2
+
+if [[ "${file}" = "" ]]; then
+	usage
+	exit 1
+fi
+
+if [[ ! -f "${file}" ]]; then
+	echo "${file}: No such file"
+	exit 1
+fi
+
+if [[ ! "${ver}" =~ v[0-9]+.[0-9]+ ]]; then
+	echo "${ver}: Invalid version number"
+	echo "Example of valid version number: v2.0"
+	exit 1
+fi
+
+latest_ver_path=$(find ${xforms_dir} -type d -and -name 'v*.*' | sort -V | tail -n 1)
+latest_ver=${latest_ver_path##*/}
+outfile=${file%.xml}-${latest_ver}.xml
+if [[ -f ${outfile} ]]; then
+	echo -n "${outfile} already exists. Override? [y/N]: "
+	read
+	if [[ ${REPLY} != "y" && ${REPLY} != "Y" ]]; then
+		echo "Stop."
+		exit 1
+	fi
+fi
+cp ${file} ${outfile}
+
+IFS='.' read -ra current_version <<<"${ver:1}"
+xforms_dir=$(dirname ${BASH_SOURCE})/xforms
+find ${xforms_dir} -type d -and -name 'v*.*' | sort -V | while read dir; do
+	target_version_string="${dir##*/v}"
+	IFS='.' read -ra target_version <<<"${target_version_string}"
+	if ((${current_version[0]} < ${target_version[0]} || (${current_version[0]} == ${target_version[0]} && ${current_version[1]} < ${target_version[1]}))); then
+		echo "Upgrade to ${target_version_string}"
+		find ${dir} -name *.xsl | sort | while read xform; do
+			echo -e "\tApplying ${xform}"
+			xsltproc -o ${outfile} ${xform} ${outfile}
```

Comment:
I would add a check at the beginning of the script to check that both `xsltproc` and `xmllint` are installed. (`if [[ -x`)

Reply:
Should we check for any potential error while running this?

Reply:
Will add both checks in v2.